src/mod/eventsystem/event_system.go

```diff
@@ -86,31 +86,78 @@ func (em *eventManager) UnregisterSubscriber(listenerID ListenerID) error {
 	return nil
 }
 
+// EmitToSubscribersAnd dispatches an event to the specific listeners in addition to the events subscribers.
+//
+// The primary use-case of this function is for plugin-to-plugin communication
+func (em *eventManager) EmitToSubscribersAnd(listenerIDs []ListenerID, payload events.EventPayload) {
+	eventName := payload.GetName()
+
+	if len(listenerIDs) == 0 {
+		return // No listeners specified
+	}
+
+	// Create the event
+	event := events.Event{
+		Name:      eventName,
+		Timestamp: time.Now().Unix(),
+		UUID:      uuid.New().String(),
+		Data:      payload,
+	}
+
+	// Dispatch to all specified listeners asynchronously
+	em.emitTo(listenerIDs, event)
+
+	// Also emit to all subscribers of the event as usual
+	em.mutex.RLock()
+	defer em.mutex.RUnlock()
+	subscribers, exists := em.subscriptions[eventName]
+	if !exists || len(subscribers) == 0 {
+		return // No subscribers
+	}
+	em.emitTo(subscribers, event)
+}
```

Comment:
in addition to listeners subscribed to this event, also dispatch to the specified plugins.

This is the key function that will eventually enable plugin<->plugin communication since CustomEvent events will specify their recipients.